src/harness/core/strings.ts

```diff
@@ -0,0 +1,130 @@
+namespace core {
+    export function padLeft(text: string, size: number, ch = " "): string {
+        while (text.length < size) text = ch + text;
+        return text;
+    }
+
+    export function padRight(text: string, size: number, ch = " "): string {
+        while (text.length < size) text += ch;
+        return text;
+    }
+
+    export function getByteOrderMark(text: string): string {
+        const length = getByteOrderMarkLength(text);
+        return length > 0 ? text.slice(0, length) : "";
+    }
+
+    export function getByteOrderMarkLength(text: string): number {
+        if (text.length >= 1) {
+            const ch0 = text.charCodeAt(0);
+            if (ch0 === 0xfeff) return 1;
+            if (ch0 === 0xfe) return text.length >= 2 && text.charCodeAt(1) === 0xff ? 2 : 0;
+            if (ch0 === 0xff) return text.length >= 2 && text.charCodeAt(1) === 0xfe ? 2 : 0;
+            if (ch0 === 0xef) return text.length >= 3 && text.charCodeAt(1) === 0xbb && text.charCodeAt(2) === 0xbf ? 3 : 0;
+        }
+        return 0;
+    }
+
+    export function removeByteOrderMark(text: string): string {
+        const length = getByteOrderMarkLength(text);
+        return length ? text.slice(length) : text;
+    }
+
+    export function addUTF8ByteOrderMark(text: string) {
+        return getByteOrderMarkLength(text) === 0 ? "\u00EF\u00BB\u00BF" + text : text;
+    }
+
+    function splitLinesWorker(text: string, lineStarts: number[] | undefined, lines: string[] | undefined, removeEmptyElements: boolean) {
+        let pos = 0;
+        let end = 0;
+        let lineStart = 0;
+        let nonWhiteSpace = false;
+        while (pos < text.length) {
+            const ch = text.charCodeAt(pos);
+            end = pos;
+            pos++;
+            switch (ch) {
+                // LineTerminator
+                case 0x000d: // <CR> carriage return
+                    if (pos < text.length && text.charCodeAt(pos) === 0x000a) {
+                        pos++;
+                    }
+                    // falls through
+
+                case 0x000a: // <LF> line feed
+                case 0x2028: // <LS> line separator
+                case 0x2029: // <PS> paragraph separator
+                    if (lineStarts) {
+                        lineStarts.push(lineStart);
+                    }
+                    if (lines && (!removeEmptyElements || nonWhiteSpace)) {
+                        lines.push(text.slice(lineStart, end));
+                    }
+                    lineStart = pos;
+                    nonWhiteSpace = false;
+                    break;
+
+                // WhiteSpace
+                case 0x0009: // <TAB> tab

```

Comment:
isWhiteSpaceSingleLine?

Reply:
This function will be removed. See https://github.com/Microsoft/TypeScript/pull/20763#discussion_r183806003 for more information.